### ISSUE TITLE
[Bug]: snowflake_object_parameter on warehouse constantly seen as modified

### ISSUE THREAD

**Issue by danielefrigo** [Snowflake-Labs/terraform-provider-snowflake#2910]
### Terraform CLI Version

1.9.1

### Terraform Provider Version

0.92.0

### Terraform Configuration

```terraform
I created a warehouse and modified a specific parameter on it:

resource snowflake_warehouse test {
  name = "TEST"
  warehouse_size = "xsmall"
  initially_suspended = true
  auto_resume         = true
  auto_suspend        = 60
}

resource "snowflake_object_parameter" "test" {
  key = "STATEMENT_TIMEOUT_IN_SECONDS"
  value = 600
  object_type = "WAREHOUSE"
  object_identifier {
    name = snowflake_warehouse.test.name
  }
}

Everything works fine, but every time I run terraform the object_parameter is detected as modified and thus update.
```

### Category

category:resource

### Object type(s)

resource:object_parameter

### Expected Behavior

Since the parameter value is unchanged, nothing should happen

### Actual Behavior

The parameter is updated at any terraform apply

### Steps to Reproduce

1. create a snowflake_warehouse
2. create a snowflake_object_parameter connected to the above warehouse
3. run terraform apply
4. run terraform plan --> parameter change is planned over and over again

### How much impact is this issue causing?

Low

### Would you like to implement a fix?

- [ ] Yeah, I'll take it 😎

**Comment by sfc-gh-jmichalak** [comment]
Hey @danielefrigo . Thanks for reaching out to us.

The warehouse resource is being redesigned as part of https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/ROADMAP.md#preparing-essential-ga-objects-for-the-provider-v1. This is already reworked, but not released yet - should happen very soon.

**Comment by sfc-gh-asawicki** [comment]
Hey @danielefrigo. One additional comment from my side.

With the current resources redesign, we are slowly moving away from allowing the same parameter configuration in different places. This is the exact problem you observe here. `statement_timeout_in_seconds` is configured in your example in two places:
1. `snowflake_object_parameter` (value: 600)
2. `snowflake_warehouse` directly (default value on the resource level: 172800)

So essentially, you want to configure the same parameter in two places to different values, so continuous diff is expected.

In the current version, you can just set `statement_timeout_in_seconds` on the warehouse level and get rid of the `snowflake_object_parameter`.

In the upcoming v0.93.0, the behavior will be a bit different because we are getting rid of the provider's internal defaults (we leave it to Snowflake). The current example will work exactly the same, though. The reason for that is, that the new warehouse implementation would detect any drift to the parameter, including the level on which the parameter is set, so:
- not setting the parameter on the warehouse level will mean that Snowflake will set the parameter to the account/snowflake default value (depending on your account setup)
- setting in the `snowflake_object_parameter` will change the level to object, so the warehouse resource will discover a drift (similarly to having two different values in your initial example).

Long story short do not use both objects simultaneously:
- we recommend handling the parameters on the object resource level; we will add missing ones to all the resources during the rework (they are there for warehouse already)
- we are still considering leaving the `snowflake_object_parameter` resource but for the cases where you don't want to manage the whole object but only its parameters

Also, you can check the upcoming migration notes (there is a long warehous-dedicated section: https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0920--v0930).
